src/GenerateYotiLetterHandler.ts

```diff
@@ -2,35 +2,63 @@ import { Logger } from "@aws-lambda-powertools/logger";
 import { Metrics } from "@aws-lambda-powertools/metrics";
 import { LambdaInterface } from "@aws-lambda-powertools/commons";
 import { Constants } from "./utils/Constants";
+import { AppError } from "./utils/AppError";
+import { HttpCodesEnum } from "./utils/HttpCodesEnum";
+import { MessageCodes } from "./models/enums/MessageCodes";
+import { getParameter } from "./utils/Config";
+import { Response } from "./utils/Response";
+import { GenerateYotiLetterProcessor } from "./services/GenerateYotiLetterProcessor";
+import { EnvironmentVariables } from "./services/EnvironmentVariables";
+import { ServicesEnum } from "./models/enums/ServicesEnum";
 
 const {
 	POWERTOOLS_METRICS_NAMESPACE = Constants.F2F_METRICS_NAMESPACE,
 	POWERTOOLS_LOG_LEVEL = Constants.DEBUG,
-	POWERTOOLS_SERVICE_NAME = "TODO",
+	POWERTOOLS_SERVICE_NAME = Constants.GENERATE_YOTI_LETTER_SVC_NAME,
 } = process.env;
 
 
-const logger = new Logger({
+export const logger = new Logger({
 	logLevel: POWERTOOLS_LOG_LEVEL,
 	serviceName: POWERTOOLS_SERVICE_NAME,
 });
 
-const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE, serviceName: POWERTOOLS_SERVICE_NAME });
+let yotiPrivateKey: string;
+
+const metrics = new Metrics({ namespace: POWERTOOLS_METRICS_NAMESPACE });
+
+export class GenerateYotiLetterHandler implements LambdaInterface {
+	private readonly environmentVariables = new EnvironmentVariables(logger, ServicesEnum.PERSON_INFO_KEY_SERVICE);
 
-class GenerateYotiLetterHandler implements LambdaInterface {
 	@metrics.logMetrics({ throwOnEmptyMetrics: false, captureColdStartMetric: true })
-	handler(event: any, context: any): any {
+	async handler(event: any, context: any): Promise<any> {
 
 		logger.setPersistentLogAttributes({});
 		logger.addContext(context);
+		try {
+			try {
+				const yotiPrivateKeyPath = this.environmentVariables.yotiKeySsmPath();
+				logger.info({ message: "Fetching key", yotiPrivateKeyPath });
```

Comment:
Can the log say exactly which key we are trying to fetch and also update the log in the catch block to mention the key name?